kernel/rtlil.cc

```diff
@@ -2373,6 +2382,24 @@ std::map<unsigned int, RTLIL::Wire*> *RTLIL::Wire::get_all_wires(void)
 }
 #endif
 
+bool RTLIL::Wire::is_parameter() const {
+	if (attributes.count("\\is_parameter")) {
+		return attributes.at("\\is_parameter").as_bool();
+	}
+	return false;
+}
+
+bool RTLIL::Wire::is_localparam() const {
+	if (attributes.count("\\is_localparam")) {
+		return attributes.at("\\is_localparam").as_bool();
+	}
+	return false;
+}
+
+bool RTLIL::Wire::is_mockup() const {
+	return is_parameter() || is_localparam();
+}
```

Comment:
There is no need to add any of those three functions or any of the many uses you add for them. I have no idea why you think you need to add hundreds of lines of code all over the place for this. The reason we have suggested the use of an attribute is so that this can be accomplished with a small non-intrusive change.